Controller/ContentTypeController.php

```diff
@@ -61,22 +63,32 @@ class ContentTypeController extends Controller
      */
     private $fieldTypeMapperRegistry;
 
+    /**
+     * @var string[]
+     */
     private $prioritizedLanguages = [];
 
+    /**
+     * @var \EzSystems\PlatformUIBundle\ApplicationConfig\Providers\Languages
+     */
+    private $languagesConfigProvider;
+
     public function __construct(
         ContentTypeService $contentTypeService,
         SearchService $searchService,
         UserService $userService,
         ActionDispatcherInterface $contentTypeGroupActionDispatcher,
         ActionDispatcherInterface $contentTypeActionDispatcher,
-        FieldTypeFormMapperRegistryInterface $fieldTypeMapperRegistry
+        FieldTypeFormMapperRegistryInterface $fieldTypeMapperRegistry,
+        LanguagesConfigProvider $languagesConfigProvider
```

Comment:
In retro you can just inject LanguageService like LanguagesConfigProvider does and load `$this->languageService->loadLanguages()` when needed, the abstraction does not provide any value here and just complicates reading the code.